Sources/UnionNotifications/iOS18NotificationView.swift

```diff
@@ -1,79 +1,132 @@
-                    .fixedSize(horizontal: false, vertical: true)
-            }
-            .padding(.vertical, 20)
-            .padding(.horizontal, 16)
 
+    var body: some View {
+        GeometryReader { geometry in
             VStack(spacing: 0) {
-                Divider()
-
-                HStack(spacing: 0) {
-                    Text("Don't Allow", bundle: .module)
-                        .font(.system(size: 17, weight: .regular))
-                        .foregroundStyle(Color.blue.opacity(0.3))
-                        .frame(maxWidth: .infinity, minHeight: 44)
-
-                    Divider()
-
-                    Button {
-                        onAllow()
-                    } label: {
-                        Text("Allow", bundle: .module)
-                            .font(.system(size: 17, weight: .medium))
-                            .foregroundStyle(Color.blue)
-                            .frame(maxWidth: .infinity, minHeight: 44)
-                    }
-                    .contentShape(.rect)
-                    .buttonStyle(UnionButtonStyle(nil) { label, isPressed in
-                        label
-                            .frame(maxWidth: .infinity, minHeight: 44)
-                            .background(
-                                Rectangle()
-                                    .fill(Color.primary.opacity(isPressed ? 0.1 : 0))
-                            )
-                    })
-                }
-                .frame(height: 44)
+                IOS18NotificationDialogContent(
+                    appName: appName,
+                    shouldEnableScroll: shouldEnableScroll,
+                    maxContentHeight: NotificationDialogConstants.maxScrollHeight
+                )
+                buttonSection
+            }
+            .background(.regularMaterial)
+            .frame(width: dialogWidth(proxy: geometry))
+            .clipShape(.rect(cornerRadius: 16, style: .continuous))
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+        }
+    }
+    
+    private var buttonSection: some View {
+        VStack(spacing: 0) {
+            Divider()
+            
+            if shouldStackButtons {
+                verticalButtons
+            } else {
+                horizontalButtons
             }
         }
-        .background {
-            Rectangle()
-                .fill(.regularMaterial)
+    }
+    
+    private var verticalButtons: some View {
+        VStack(spacing: 0) {
+            disabledButton
+            Divider()
+            activeButton
+        }
+    }
+    
+    private var horizontalButtons: some View {
+        HStack(spacing: 0) {
+            disabledButton
+            Divider()
+            activeButton
+        }
+        .frame(height: buttonHeight)
+    }
+    
+    private var disabledButton: some View {
+        Button(action: {
+                onDontAllow?()  
+            }) {
+            Text("Don't Allow", bundle: .module)
```

Comment:
Make sure that your strings are using angled quotes. The system dialog uses these, so it's implicit in the spec.